From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 9 Oct 2015 17:18:18 +0200
Subject: core: ignore -.slice and init.scope when isolating

Otherwise, we might end up trying to isolate it away when starting user
instances.

While we are at it, also prohibit manual start/stop of these two units.

Fixes: #1507
---
 src/core/scope.c | 3 +++
 src/core/slice.c | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/src/core/scope.c b/src/core/scope.c
index 7325e36..ea7d846 100644
--- a/src/core/scope.c
+++ b/src/core/scope.c
@@ -534,6 +534,9 @@ static int scope_enumerate(Manager *m) {
         u->transient = true;
         u->default_dependencies = false;
         u->no_gc = true;
+        u->ignore_on_isolate = true;
+        u->refuse_manual_start = true;
+        u->refuse_manual_stop = true;
         SCOPE(u)->deserialized_state = SCOPE_RUNNING;
         SCOPE(u)->kill_context.kill_signal = SIGRTMIN+14;
 
diff --git a/src/core/slice.c b/src/core/slice.c
index 0630241..1542e83 100644
--- a/src/core/slice.c
+++ b/src/core/slice.c
@@ -274,6 +274,9 @@ static int slice_enumerate(Manager *m) {
 
         u->default_dependencies = false;
         u->no_gc = true;
+        u->ignore_on_isolate = true;
+        u->refuse_manual_start = true;
+        u->refuse_manual_stop = true;
         SLICE(u)->deserialized_state = SLICE_ACTIVE;
 
         if (!u->description)
